### PR TITLE
feat: replace selector with query object

### PR DESCRIPTION
**Pull request by Saul-Mirone**

### CODE REVIEW

**Comment by Saul-Mirone** [comment]
* **#7755** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7755?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Saul-Mirone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by doouding** [comment]
Is there any doc on how to use the new query object?

**Comment by Saul-Mirone** [comment]
> Is there any doc on how to use the new query object?

I plan to write something in next pr.

File: packages/framework/store/src/store/doc/query.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import type { BlockModel } from '../../schema/index.js';
+import type { Block } from './block/index.js';
+
+import { BlockViewType } from './doc.js';
+
+export type QueryMatch = {
+  id?: string;
+  flavour?: string;
+  props?: Record<string, unknown>;
+  viewType: BlockViewType;
```
Comment (EYHN) [inline packages/framework/store/src/store/doc/query.ts:12]: just wonder why we need to define viewtype in query. If there are only include and not_include, the design here will be simpler.

Comment (Saul-Mirone) [reply packages/framework/store/src/store/doc/query.ts:12]: by pass is needed for render a block without render the view of its ancestors

**Pull request merged**